themes/bootstrap3/templates/RecordDriver/DefaultRecord/result-list.phtml

```diff
@@ -21,7 +22,7 @@
   <div class="media-body">
     <div class="result-body">
       <h2>
-        <a href="<?=$this->escapeHtmlAttr($recordLinker->getUrl($this->driver))?>" class="title getFull" data-view="<?=isset($this->params) ? $this->params->getOptions()->getListViewOption() : 'list' ?>">
+        <a href="<?=$this->escapeHtmlAttr($recordLinker->getUrl($this->driver, ['searchOrigin' => $this->get('searchOrigin')]))?>" class="title getFull" data-view="<?=isset($this->params) ? $this->params->getOptions()->getListViewOption() : 'list' ?>">
```

Comment:
Should this be simply `$this->searchOrigin` or (if you agree with my earlier suggestion) `$this->layout()->searchOrigin` instead of using the `get()` method (which is not something we typically do elsewhere in our templates)?